src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php

```diff
@@ -710,11 +710,19 @@ private function resolveServices($value, $file, $isParameter = false)
                 }
             }
             if ('tagged' === $value->getTag()) {
-                if (!\is_string($argument) || !$argument) {
-                    throw new InvalidArgumentException(sprintf('"!tagged" tag only accepts non empty string in "%s".', $file));
+                if (\is_string($argument) && $argument) {
+                    return new TaggedIteratorArgument($argument);
                 }
 
-                return new TaggedIteratorArgument($argument);
+                if (\is_array($argument) && isset($argument['tag']) && $argument['tag']) {
+                    if (array_diff(array_keys($argument), ['tag', 'index_by', 'default_index_method'])) {
+                        throw new InvalidArgumentException('"!tagged" tag contains unsupported keys. Supported are: "tag, index_by, default_index_method".');
```

Comment:
`'"!tagged" tag contains unsupported key "%s"; supported ones are "tag", "index_by" and "default_index_method".'`